### PR TITLE
Add containsDuplicates to check whether a collection contains duplicates

### PR DESCRIPTION
**Pull request by Vyax**
## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Sources/Extensions/SwiftStdlib/CollectionExtensions.swift [hunk 1]
```diff
@@ -188,3 +188,15 @@ public extension Collection where Element: FloatingPoint {
     }
 
 }
+
+// MARK: - Methods (Hashable)
```
Comment (LucianoPAlmeida) [inline Sources/Extensions/SwiftStdlib/CollectionExtensions.swift]: I think here it should be 'Properties' instead of 'Methods' :))

File: Tests/SwiftStdlibTests/CollectionExtensionsTests.swift [hunk 2]
```diff
@@ -151,4 +151,9 @@ final class CollectionExtensionsTests: XCTestCase {
         XCTAssertEqual([Double]().average(), 0)
     }
 
+    func testContainsDuplicates() {
+        XCTAssert(["a", "b", "b", "c"].containsDuplicates)
+        XCTAssertFalse(["a", "b", "c", "d"].containsDuplicates)
```
Comment (BennX) [inline Tests/SwiftStdlibTests/CollectionExtensionsTests.swift]: I'm missing the edge condition with an empty array.

Comment (Vyax) [reply Tests/SwiftStdlibTests/CollectionExtensionsTests.swift]: Okay, will be added.

File: Sources/Extensions/SwiftStdlib/CollectionExtensions.swift [hunk 3]
```diff
@@ -188,3 +188,15 @@ public extension Collection where Element: FloatingPoint {
     }
 
 }
+
+// MARK: - Properties (Hashable)
+public extension Collection where Element: Hashable {
```
Comment (guykogus) [inline Sources/Extensions/SwiftStdlib/CollectionExtensions.swift]: Can you put this a level up and make it an extension of `Sequence`?

Comment (guykogus) [reply Sources/Extensions/SwiftStdlib/CollectionExtensions.swift]: Actually, I just realised `Sequence` doesn't have a `count`. You could still put it there, but you'd need to manually add each element to the set, and check before if it exists. e.g.
extension Sequence where Element: Hashable {
```Swift
extension Sequence where Element: Hashable {
    func containsDuplicates() -> Bool {
        var set = Set<Element>()
        for element in self {
            if !set.insert(element).inserted {
                return true
            }
        }
        return false
    }
}
```

File: Sources/Extensions/SwiftStdlib/CollectionExtensions.swift [hunk 4]
```diff
@@ -188,3 +188,15 @@ public extension Collection where Element: FloatingPoint {
...
+
+    /// SwifterSwift: Checks whether a collection contains duplicates.
+    ///
+    /// - Returns: true if the receiver contains duplicates.
+    public var containsDuplicates: Bool {
```
Comment (guykogus) [inline Sources/Extensions/SwiftStdlib/CollectionExtensions.swift]: This should be a `func`, since it won't have constant runtime.

**Comment by SD10** [comment]
@guykogus @BennX This should be good to go if anyone else wants to review
